src/app.jsx

```diff
@@ -32,9 +32,10 @@ export default class App extends React.Component {
         }}
       />
       <Viewer 
-        onOpen={this.onOpen.bind(this)}
+        onOpen={this.onOpen.bind(this)} ref={
```

Comment:
Do you know how this ref attribute works? I know I put it in first, but that was just copying an example, be good if you can explain it to me!

Reply:
It's to bind `this` - which is not bound in class methods by default, can be bypassed by using arrow functions or some new syntax, https://reactjs.org/docs/handling-events.html

Reply:
No, not the `.bind(this)`, the `ref = {viewer => {this.viewer = viewer;}}`. I can see that it gives the parent component access to the child component, but I don't understand what the the `ref` attribute is and why this callback needs to be assigned to it

Reply:
https://reactjs.org/docs/refs-and-the-dom.html#callback-refs to access the viewer DOM node I presume